<?xml version="1.0" ?>
<robot name="rescue">

<material name="body">
    <color rgba="0 0 1 1"/>
</material>
<material name="flipper">
    <color rgba="0 0.8 1 1"/>
</material>
<material name="arm">
    <color rgba="0 1 0 1"/>
</material>
<material name="wrist">
    <color rgba="1 1 0 1"/>
</material>
<material name="gripper">
    <color rgba="1 1 1 1"/>
</material>

<link name="base_link">
</link>
<joint name="body_offset" type="fixed">
    <parent link="base_link"/>
    <child link="main_body"/>
    <origin rpy="1.57075 0 0" xyz="-0.105 -0.17 0.15"/>
</joint>

<link name="main_body">
  <inertial>
    <origin rpy="0 0 0" xyz="0.10362958128296546 -0.018518589851868602 -0.11928098661532627"/>
    <mass value="125.39802798594614"/>
    <inertia ixx="0.710702" ixy="0.02898" ixz="-0.000172" iyy="3.591341" iyz="0.000124" izz="3.324467"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <geometry>
      <mesh filename="package://state_publisher/meshes/base_link_m-binary.stl"/>
    </geometry>
    <material name="body"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/base_link_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="right_belt_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.22743918683014153 -0.0023270038235164714 0.025603511116359286"/>
    <mass value="50.63722379743233"/>
    <inertia ixx="0.196333" ixy="-0.0" ixz="0.0" iyy="1.451152" iyz="-7.1e-05" izz="1.62557"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.010522"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/right_belt_1_m-binary.stl"/>
    </geometry>
    <material name="body"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.010522"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/right_belt_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="left_belt_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.22743918683022915 -0.002327003823516492 -0.025604306397982307"/>
    <mass value="50.637223797432235"/>
    <inertia ixx="0.196333" ixy="-0.0" ixz="-0.0" iyy="1.451152" iyz="7.1e-05" izz="1.62557"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.228023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/left_belt_1_m-binary.stl"/>
    </geometry>
    <material name="body"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.228023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/left_belt_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="flipper_rear_right_1">
  <inertial>
    <origin rpy="0 0 0" xyz="1.8667320490772799e-07 0.06327889494871872 0.017692078578064464"/>
    <mass value="11.639210054919547"/>
    <inertia ixx="0.081135" ixy="-0.0" ixz="-0.0" iyy="0.020636" iyz="3.3e-05" izz="0.099255"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 -0.044478"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_rear_right_1_m-binary.stl"/>
    </geometry>
    <material name="flipper"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 -0.044478"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_rear_right_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="flipper_front_right_1">
  <inertial>
    <origin rpy="0 0 0" xyz="1.86673220603506e-07 0.06327889494871752 0.017692078578063312"/>
    <mass value="11.639210054919547"/>
    <inertia ixx="0.081135" ixy="0.0" ixz="-0.0" iyy="0.020636" iyz="3.3e-05" izz="0.099255"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.333459 0.029614 -0.044478"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_front_right_1_m-binary.stl"/>
    </geometry>
    <material name="flipper"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.333459 0.029614 -0.044478"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_front_right_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="flipper_rear_left_1">
  <inertial>
    <origin rpy="0 0 0" xyz="1.866732055877396e-07 0.06327889494871869 -0.017692873859686542"/>
    <mass value="11.639210054919547"/>
    <inertia ixx="0.081135" ixy="-0.0" ixz="-0.0" iyy="0.020636" iyz="-3.3e-05" izz="0.099255"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.283023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_rear_left_1_m-binary.stl"/>
    </geometry>
    <material name="flipper"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0.121419 0.029614 0.283023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_rear_left_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="flipper_front_left_1">
  <inertial>
    <origin rpy="0 0 0" xyz="1.8667322182475132e-07 0.06327889494871758 -0.01769287385968804"/>
    <mass value="11.639210054919547"/>
    <inertia ixx="0.081135" ixy="-0.0" ixz="-0.0" iyy="0.020636" iyz="-3.3e-05" izz="0.099255"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.333459 0.029614 0.283023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_front_left_1_m-binary.stl"/>
    </geometry>
    <material name="flipper"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.333459 0.029614 0.283023"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/flipper_front_left_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="arm_base_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.0015250817864826405 0.036972179347615125 -0.009123390160887465"/>
    <mass value="9.318674950833845"/>
    <inertia ixx="0.020384" ixy="-0.000132" ixz="-0.000105" iyy="0.026478" iyz="-4.7e-05" izz="0.017031"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.313459 -0.047886 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_base_1_m-binary.stl"/>
    </geometry>
    <material name="arm"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.313459 -0.047886 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_base_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="arm_lower_1">
  <inertial>
    <origin rpy="0 0 0" xyz="-0.33514903590256295 0.004913441536999602 0.012839710067103355"/>
    <mass value="3.211993258458033"/>
    <inertia ixx="0.004374" ixy="0.00084" ixz="0.000477" iyy="0.029637" iyz="0.000198" izz="0.026449"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.299027 -0.136079 0.125273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_lower_1_m-binary.stl"/>
    </geometry>
    <material name="arm"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.299027 -0.136079 0.125273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_lower_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="arm_upper_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.22451215756009318 1.3010842617455776e-05 0.006000101925213053"/>
    <mass value="1.1170241298535344"/>
    <inertia ixx="0.000146" ixy="0.0" ixz="-0.0" iyy="0.010141" iyz="0.0" izz="0.01012"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0.098995 -0.175902 0.125273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_upper_1_m-binary.stl"/>
    </geometry>
    <material name="arm"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0.098995 -0.175902 0.125273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/arm_upper_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="wrist_yaw_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.04441576596725069 4.622440790394422e-05 0.019000101925213286"/>
    <mass value="0.5097860192503678"/>
    <inertia ixx="0.000112" ixy="-0.0" ixz="-0.0" iyy="0.000264" iyz="0.0" izz="0.000208"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.216685 -0.175944 0.138273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/wrist_yaw_1_m-binary.stl"/>
    </geometry>
    <material name="wrist"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.216685 -0.175944 0.138273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/wrist_yaw_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="wrist_pitch_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.037373819563222666 -0.0061108662348390574 1.0192522521823566e-07"/>
    <mass value="0.39341698942281567"/>
    <inertia ixx="0.000118" ixy="-1.1e-05" ixz="-0.0" iyy="9.8e-05" iyz="0.0" izz="0.000176"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.27322 -0.190503 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/wrist_pitch_1_m-binary.stl"/>
    </geometry>
    <material name="wrist"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.27322 -0.190503 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/wrist_pitch_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<link name="gripper_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.02583338452773354 2.6768630369794488e-05 4.48856803267339e-07"/>
    <mass value="1.7001261051527197"/>
    <inertia ixx="0.004444" ixy="-0.0" ixz="-0.0" iyy="0.004742" iyz="-0.0" izz="0.001496"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.332235 -0.176065 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/gripper_1_m-binary.stl"/>
    </geometry>
    <material name="gripper"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.332235 -0.176065 0.119273"/>
    <geometry>
        <mesh filename="package://state_publisher/meshes/gripper_1_m-binary.stl"/>
    </geometry>
  </collision>
</link>
<joint name="Rigid1" type="fixed">
  <origin rpy="0 0 0" xyz="-0.121419 -0.029614 -0.010522"/>
  <parent link="main_body"/>
  <child link="right_belt_1"/>
</joint>
<transmission name="Rigid1_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Rigid1">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="Rigid1_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="Rigid2" type="fixed">
  <origin rpy="0 0 0" xyz="-0.121419 -0.029614 -0.228023"/>
  <parent link="main_body"/>
  <child link="left_belt_1"/>
</joint>
<transmission name="Rigid2_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Rigid2">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="Rigid2_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="flipper_rear" type="revolute">
  <origin rpy="0 0 0" xyz="0.0 0.0 0.055"/>
  <parent link="right_belt_1"/>
  <child link="flipper_rear_right_1"/>
  <axis xyz="0.0 0.0 0.01"/>
  <limit effort="100" lower="-1.570796" upper="3.316126" velocity="100"/>
</joint>
<transmission name="flipper_rear_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="flipper_rear">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="flipper_rear_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="flipper_front" type="revolute">
  <origin rpy="0 0 0" xyz="0.454878 0.0 0.055"/>
  <parent link="right_belt_1"/>
  <child link="flipper_front_right_1"/>
  <axis xyz="0.0 0.0 0.01"/>
  <limit effort="100" lower="-3.316126" upper="1.570796" velocity="100"/>
</joint>
<transmission name="flipper_front_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="flipper_rear">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="flipper_front_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="Rigid5" type="fixed">
  <origin rpy="0 0 0" xyz="0.0 0.0 -0.327501"/>
  <parent link="flipper_rear_right_1"/>
  <child link="flipper_rear_left_1"/>
</joint>
<transmission name="Rigid5_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Rigid5">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="Rigid5_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="Rigid6" type="fixed">
  <origin rpy="0 0 0" xyz="0.0 0.0 -0.327501"/>
  <parent link="flipper_front_right_1"/>
  <child link="flipper_front_left_1"/>
</joint>
<transmission name="Rigid6_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Rigid6">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="Rigid6_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="arm1" type="revolute">
  <origin rpy="0 0 0" xyz="0.313459 0.047886 -0.119273"/>
  <parent link="main_body"/>
  <child link="arm_base_1"/>
  <axis xyz="-0.0 -0.01 -0.0"/>
  <limit effort="100" lower="-2.356194" upper="2.356194" velocity="100"/>
</joint>
<transmission name="arm1_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="arm1">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="arm1_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="arm2" type="revolute">
  <origin rpy="0 0 0" xyz="-0.014432 0.088193 -0.006"/>
  <parent link="arm_base_1"/>
  <child link="arm_lower_1"/>
  <axis xyz="0.0 0.0 0.01"/>
  <limit effort="100" lower="-3.141593" upper="0.0" velocity="100"/>
</joint>
<transmission name="arm2_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="arm2">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="arm2_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="arm3" type="revolute">
  <origin rpy="0 0 0" xyz="-0.398022 0.039823 0.0"/>
  <parent link="arm_lower_1"/>
  <child link="arm_upper_1"/>
  <axis xyz="0.0 -0.0 0.01"/>
  <limit effort="100" lower="0.0" upper="3.141593" velocity="100"/>
</joint>
<transmission name="arm3_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="arm3">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="arm3_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="wrist1" type="continuous">
  <origin rpy="0 0 0" xyz="0.31568 4.2e-05 -0.013"/>
  <parent link="arm_upper_1"/>
  <child link="wrist_yaw_1"/>
  <axis xyz="0.0 0.0 0.01"/>
</joint>
<transmission name="writst1_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="wrist1">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="wrist1_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="wrist2" type="continuous">
  <origin rpy="0 0 0" xyz="0.056535 0.014559 0.019"/>
  <parent link="wrist_yaw_1"/>
  <child link="wrist_pitch_1"/>
  <axis xyz="1e-05 -0.01 -0.0"/>
</joint>
<transmission name="wrist2_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="wrist2">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="wrist2_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<joint name="wrist3" type="continuous">
  <origin rpy="0 0 0" xyz="0.059015 -0.014438 0.0"/>
  <parent link="wrist_pitch_1"/>
  <child link="gripper_1"/>
  <axis xyz="0.01 1e-05 -0.0"/>
</joint>
<transmission name="wrist3_tran">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="wrist3">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
  </joint>
  <actuator name="wrist3_actr">
    <hardwareInterface>PositionJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>
<gazebo>
  <plugin filename="libgazebo_ros_control.so" name="control"/>
</gazebo>
</robot>
